hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java

```diff
@@ -3933,4 +3937,25 @@ public void onConfigurationChange(Configuration conf) {
     super.onConfigurationChange(conf);
     setReloadableGuardrails(conf);
   }
+
+  @Override
+  public GetPrefetchedFilesListResponse getPrefetchedFilesList(RpcController controller,
+    GetPrefetchedFilesListRequest request) throws ServiceException {
+    GetPrefetchedFilesListResponse.Builder responseBuilder =
+      GetPrefetchedFilesListResponse.newBuilder();
+
+    List<String> fullyCachedFiles = new ArrayList<>();
+    server.getBlockCache().ifPresent(blockCache -> {
+      if (blockCache instanceof CombinedBlockCache) {
+        BlockCache l2 = ((CombinedBlockCache) blockCache).getSecondLevelCache();
+        if (l2 instanceof BucketCache) {
+          if (((BucketCache) l2).isCachePersistenceEnabled()) {
```

Comment:
Use the static method BucketCache.getBucketCacheFromCacheConfig for finding the bucket cache instance.

Reply:
Thanks for the feedback @wchevreuil, However, my idea is that we have RegionServer.getBlockCache(), so I added a getter (getBucketCacheIfPresent()) for the bucket cache as a property of the RS. 
WDYT?

Reply:
It's fine to add the additional getter, but we should not duplicate code. This whole implementation is identical to BucketCache.getBucketCacheFromCacheConfig, so you should just call it inside your method.